### ISSUE TITLE
Run scaling experiments with both increasing model and dataset size

### ISSUE THREAD

**Issue by mcw92** [Helmholtz-AI-Energy/special-couscous#24]
In our current weak-scaling series, we scale the problem size via the model dimension, i.e., by increasing the number of trees with the number of nodes. The dataset to be classified is the same for each parallelization level. It is generated redundantly on each node and split into globally shared train and test sets afterwards. In this way, we can analyze how increasing the model size helps with classifying the same single dataset.

In addition, we plan an analogous experiment series, where we scale in both the model and data dimension simultaneously: When adding new nodes, each node generates its own local train set from the same global distribution (i.e., using the same hyperparameters in `make_classification` as for the weak-scaling experiments except for a locally unique seed). The number of samples in these local train sets must equal the number of samples in the weak-scaling train set after the split, i.e., `train_split * n_samples`. Note that the test set still needs to be shared across all nodes to finally evaluate the distributed global model. It can be generated redundantly on each node from the same distribution, i.e., again using the same hyperparaters except for a different but shared seed. The number of samples should be `(1 - train_split) * n_samples`.

This means each node generates
- a locally unique train set of size `train_split * n_samples` using a locally unique seed.
- the globally shared test set of size `(1 - train_split) * n_samples` using a globally shared seed.

All generated synthetic datasets must be **from the same distribution**, i.e., use the same hyperparameters in `make_classification` (except for number of samples and seed).
